src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -647,23 +646,51 @@ def get_git_base_dir():
                 .strip()
             )
 
-        dependencies = ["promptflow-devkit", "promptflow-core", "promptflow-tracing"]
-        # get promptflow-** required and extra packages
-        extra_packages = []
-        required_packages = []
-        for package in dependencies:
-            with open(get_git_base_dir() / "src" / package / "pyproject.toml", "rb") as file:
-                data = toml.load(file)
-            extras = data.get("tool", {}).get("poetry", {}).get("extras", {})
-            for _, package in extras.items():
-                extra_packages.extend(package)
-            requires = data.get("tool", {}).get("poetry", {}).get("dependencies", [])
-            for package, _ in requires.items():
-                required_packages.append(package)
+        def is_tool(name):
+            """Check whether `name` is on PATH and marked as executable."""
+
+            # from whichcraft import which
+            from shutil import which
+
+            return which(name) is not None
 
+        def get_package_dependencies(package_name_list):
+            dependencies = []
+            for package_name in package_name_list:
+                if is_tool("conda"):
+                    result = subprocess.run(
+                        "conda activate root | pip show {}".format(package_name), shell=True, stdout=subprocess.PIPE
+                    )
+                else:
+                    result = subprocess.run(["pip", "show", package_name], stdout=subprocess.PIPE)
+                lines = result.stdout.decode("utf-8", errors="ignore").splitlines()
+                for line in lines:
+                    if line.startswith("Requires"):
+                        dependency = line.split(": ")[1].split(", ")
+                        if dependency != [""]:
+                            dependencies.extend(dependency)
+                        break
+
+            dependencies = [dependency for dependency in dependencies if not dependency.startswith("promptflow")]
+            return dependencies
+
+        dependencies = ["promptflow-devkit", "promptflow-core", "promptflow-tracing"]
+        # get promptflow-** required packages
+        required_packages = get_package_dependencies(dependencies)
+        # hard-code promptflow-** required extra packages for now
+        extra_packages = [
+            "pyarrow",
+            "pyinstaller",
+            "streamlit",
+            "streamlit-quill",
+            "bs4",
+            "fastapi",
+            "azure-identity",
```

Comment:
why do we need the azure ones?

Reply:
it's promptflow core extra dependency。 Not sure whether we need this in executable
![image](https://github.com/microsoft/promptflow/assets/26239730/221ec636-8984-4ae3-a11f-6d7ea39e1c8d)